[KOGITO-8305] - [KSW-Guides] Deploying on OpenShift Clusters (#346)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/cloud/common/_common_proc_deploy_kubectl_oc.adoc

> You can also use `{command_line_tool}` command-line interface and plain Kubernetes objects to deploy your workflow application. +

Author: kube and oc, if we drop one will need to drop both :)

And, instead of creating `knative` `yaml|json` descriptors, you can leverage the Quarkus Kubernetes extension and Kogito Knative add-on to generate the descriptors.

.Prerequisites
* Kogito Workflow application ready to use.

> * `{command_line_tool}` command-line tool is installed.

Author: updated

----
<dependency>
  <groupId>org.kie.kogito</groupId>
  <artifactId>kogito-addons-quarkus-knative-eventing</artifactId>
</dependency>
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>{quarkus-k8s-plugin}</artifactId>
</dependency>
----
Gradle::
+
[source,shell,subs="attributes+"]
----
quarkus-kubernetes 'io.quarkus:{quarkus-k8s-plugin}:{quarkus_version}'
quarkus-kubernetes 'org.kie.kogito:kogito-addons-quarkus-knative-eventing:{page-component-version}'
----
Quarks CLI::
+
[source,shell,subs="attributes+"]
----
quarkus ext add org.kie.kogito:kogito-addons-quarkus-knative-eventing quarkus-openshift{page-component-version}'
----
====
--

. To generate the `knative` `yaml|json` descriptors, set the following properties in the `application.properties`
file of your workflow application:
+
--
.System properties to generate knative descriptors
[source,shell,subs="attributes+"]
----
quarkus.kubernetes.deployment-target=knative
quarkus.knative.name=hello-workflow
----
--

. Build your workflow application.
+
--
Once you have built your application, you can find the generated descriptors files in the `target/kubernetes` directory:

* `knative.json`
* `knative.yml`

[IMPORTANT]
====

> The image used in this section was already built in the previous topic, if you don't have done it yet, please

Reviewer: ```suggestion
The image used in this section was already built in the previous topic, if you do not have done it yet, you can take a look at xref:cloud/quarkus/build-workflow-image-with-quarkus-cli.adoc[Build Workflow Image with Quarkus CLI].
```

> take a look xref:cloud/quarkus/build-workflow-image-with-quarkus-cli.adoc[Build Workflow Image with Quarkus CLI].

Reviewer 2: ```suggestion
```

    app.quarkus.io/build-timestamp: 2022-06-23 - 23:53:38 +0000
  labels:
    app.kubernetes.io/name: hello-workflow
  name: hello-workflow
spec:
  template:
    metadata:
      labels:
        app.kubernetes.io/name: hello-workflow
    spec:
      containerConcurrency: 0
      containers:
        - image: {k8s_registry}/{default_namespace}/serverless-workflow-greeting-quarkus:1.0-native
          livenessProbe:
            failureThreshold: 3
            httpGet:
              path: /q/health/live
              port: null
              scheme: HTTP
            initialDelaySeconds: 0
            periodSeconds: 30
            successThreshold: 1
            timeoutSeconds: 10
          name: hello-workflow
          ports:
            - containerPort: 8080
              name: http1
              protocol: TCP
          readinessProbe:
            failureThreshold: 3
            httpGet:
              path: /q/health/ready
              port: null
              scheme: HTTP
            initialDelaySeconds: 0
            periodSeconds: 30
            successThreshold: 1
            timeoutSeconds: 10
----

[NOTE]
====

> Once the files are generated, you might be required to add the `imagePullPolicy` manually before deploying the workflow application.

Reviewer: ```suggestion
Once the files are generated, you must add the `imagePullPolicy` manually before deploying the workflow application.
```

====
--

> . Enter the following command to deploy the workflow application using `kubectl`:

Reviewer 2: ```suggestion
. Enter the following command to deploy the workflow application using `{command_line_tool}`:
```
